### PR TITLE
OpenExperience (Certified Connectors)

### PR DESCRIPTION
**Pull request by oxTomaszJ**
---
#### Important: Our certification engineer, Srikanth, is currently on emergency leave. Please expect a delay with your PR. He will return January 31st, 2022. We apologize for any inconveniences.

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Comment by oxTomaszJ** [comment]
I am unable to add a label

File: certified-connectors/OpenExperience/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,249 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Open Experience",
+    "description": "This is an Open Experience connector, which enables you to integrate the Open Experience services with your flows",
```
Comment (sriyen-msft) [inline certified-connectors/OpenExperience/apiDefinition.swagger.json:5]: @oxTomaszJ The given description is not sufficient, please provide a detailed description of your service and connector.

Docs - [Write a description for your connector](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#write-a-description-for-your-connector)

File: certified-connectors/OpenExperience/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/connector/createProject": {
+      "post": {
+        "responses": {
+          "default": {
```
Comment (sriyen-msft) [inline certified-connectors/OpenExperience/apiDefinition.swagger.json:28]: @oxTomaszJ Default responses are not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide valid expected response codes with schema definitions for all the operations in the swagger.

File: certified-connectors/OpenExperience/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+            "description": "default",
+            "schema": {}
```
Comment (sriyen-msft) [inline certified-connectors/OpenExperience/apiDefinition.swagger.json:30]: @oxTomaszJ Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

File: certified-connectors/OpenExperience/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+          {
+            "name": "Project Settings",
+            "in": "body",
+            "required": false,
+            "schema": {
+              "type": "object",
+              "properties": {
+                "customerId": {
+                  "type": "string",
+                  "description": "customerId",
```
Comment (sriyen-msft) [inline certified-connectors/OpenExperience/apiDefinition.swagger.json:45]: @oxTomaszJ All the description values are expected to be free from grammatical and spell errors. Please provide valid description values for all the properties.

File: certified-connectors/OpenExperience/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+                },
+                "address": {
+                  "type": "string",
+                  "description": "address",
+                  "title": "Address"
+                },
+                "contactPhone": {
+                  "type": "integer",
+                  "format": "int32",
+                  "description": "contactPhone",
```
Comment (sriyen-msft) [inline certified-connectors/OpenExperience/apiDefinition.swagger.json:72]: @oxTomaszJ All the description values are expected to be free from grammatical and spell errors. Please provide valid description values for all the properties.

File: certified-connectors/OpenExperience/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+                  "title": "Contact Phone"
+                },
+                "contactEmail": {
+                  "type": "string",
+                  "description": "contactEmail",
```
Comment (sriyen-msft) [inline certified-connectors/OpenExperience/apiDefinition.swagger.json:77]: @oxTomaszJ All the description values are expected to be free from grammatical and spell errors. Please provide valid description values for all the properties.

File: certified-connectors/OpenExperience/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+        "operationId": "GetAvailableCustomers",
+        "x-ms-visibility": "internal",
+        "parameters": []
+      }
+    },
+    "/connector/getUserServices": {
+      "get": {
+        "responses": {
+          "200": {
+            "description": "defaul1t",
```
Comment (sriyen-msft) [inline certified-connectors/OpenExperience/apiDefinition.swagger.json:180]: @oxTomaszJ Typo?

File: certified-connectors/OpenExperience/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+                  "items": {
+                    "type": "object",
+                    "properties": {
+                      "type": {
+                        "type": "string",
+                        "description": "type"
+                      },
+                      "serviceFullname": {
+                        "type": "string",
+                        "description": "serviceFullname"
```
Comment (sriyen-msft) [inline certified-connectors/OpenExperience/apiDefinition.swagger.json:195]: @oxTomaszJ All the description values are expected to be free from grammatical and spell errors. Please provide valid description values for all the properties.

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

File: certified-connectors/OpenExperience/apiDefinition.swagger.json (hunk 3 above)
Comment (oxTomaszJ) [reply certified-connectors/OpenExperience/apiDefinition.swagger.json:30]: If the API returns HTTP 201 (project in our services is created) it still needs to return some kind of JSON response?

**Comment by oxTomaszJ** [comment]
@sriyen-msft Could you please review changes? Thanks!

**Review by sriyen-msft — APPROVED** [review]

**Comment by sriyen-msft** [comment]
@oxTomaszJ Your connector pull request is approved and merged, you can submit your connector artifacts to the Connector certification tab in ISV Studio to move forward in the certification process, Thanks.

**Pull request merged**